search/tests/mock_search_engine.py

```diff
@@ -64,9 +65,23 @@ def value_matches(doc, field_name, field_value):
             return include_blanks
 
         # if we have a string that we are trying to process as a date object
-        if (isinstance(compare_value, basestring) and
-                (isinstance(field_value, DateRange) or isinstance(field_value, datetime))):
-            compare_value = json_date_to_datetime(compare_value)
+        if isinstance(field_value, DateRange) or isinstance(field_value, datetime):
+            if isinstance(compare_value, basestring):
+                compare_value = json_date_to_datetime(compare_value)
+
+            field_has_tz_info = False
+            if isinstance(field_value, DateRange):
+                if field_value.lower and field_value.lower.tzinfo is not None:
+                    field_has_tz_info = True
+                if field_value.upper and field_value.upper.tzinfo is not None:
+                    field_has_tz_info = True
```

Comment:
@martynjames nit: looks like it's equivalent to

    field_has_tz_info = (
        (field_value.lower and field_value.lower.tzinfo is not None) or
        (field_value.upper and field_value.upper.tzinfo is not None)
    )

or

    has_tz = lambda value = value and value.tzinfo is not None
    field_has_tz_info = has_tz(field_value.lower) or has_tz(field_value.upper) 

which I find more concise. A matter of taste though.